hydroflow/tests/surface_async.rs

```diff
@@ -1,9 +1,218 @@
+//! Surface syntax tests of asynchrony and networking.
+
 use std::cell::RefCell;
+use std::collections::HashSet;
 use std::error::Error;
+use std::net::{Ipv4Addr, SocketAddr};
 use std::rc::Rc;
 use std::time::Duration;
 
-use hydroflow::hydroflow_syntax;
+use hydroflow::scheduled::graph::Hydroflow;
+use hydroflow::{hydroflow_syntax, rassert, rassert_eq};
+use tokio::io::{AsyncBufReadExt, BufReader};
+use tokio::net::{TcpListener, TcpStream, UdpSocket};
+use tokio::task::LocalSet;
+use tokio_stream::wrappers::LinesStream;
+
+#[tokio::test]
+pub async fn test_echo_udp() -> Result<(), Box<dyn Error>> {
+    const SERVER_PORT: u16 = 8099;
+    let server_socket = SocketAddr::new(Ipv4Addr::LOCALHOST.into(), SERVER_PORT);
+
+    let local = LocalSet::new();
+
+    // Server:
+    let serv = local.spawn_local(async {
+        let socket = UdpSocket::bind((Ipv4Addr::UNSPECIFIED, SERVER_PORT))
+            .await
+            .unwrap();
+        let (send, recv) = hydroflow::util::udp_lines(socket);
+        println!("Server live!");
+
+        let seen = <Rc<RefCell<HashSet<String>>>>::default();
+        let seen_inner = Rc::clone(&seen);
+
+        let mut df: Hydroflow = hydroflow_syntax! {
+            recv_stream(recv)
+                -> map(|r| r.unwrap())
```

Comment:
is there a circumstance in which we don't `unwrap()` in Hydroflow? Seems like this should be in the helper (with error checking) not the Hydroflow syntax.

Reply:
Only certain streams return errors, in this case it would return an error if the length encoding is broken somehow. We do need a more principled design for error handling.
`-> filter_map(|r| r.ok()) ->` is another option that filters out errors quietly